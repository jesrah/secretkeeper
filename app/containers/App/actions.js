/*
 * Home Actions
 *
 * Actions change things in your application
 * Since this boilerplate uses a uni-directional data flow, specifically redux,
 * we have these actions which are the only way your application interacts with
 * your application state. This guarantees that your state is up to date and nobody
 * messes it up weirdly somewhere.
 *
 * To add a new Action:
 * 1) Import your constant
 * 2) Add a function like this:
 *    export function yourAction(var) {
 *        return { type: YOUR_ACTION_CONSTANT, var: var }
 *    }
 */

import { 
	LOAD_SECRETS,
	LOAD_SECRETS_SUCCESS,
	LOAD_SECRETS_ERROR,
	POST_SECRET
} from './constants';

/**
 * Loads all secrets, starting a request saga
 *
 * @return {object}    An action object with a type of LOAD_SECRETS
 */
export function loadSecrets() {
  return {
    type: LOAD_SECRETS,
  };
}

export function secretsLoaded(strings) {
	return {
		type: LOAD_SECRETS_SUCCESS,
		strings,
	}
}

export function secretsLoadingError(error) {
	return {
		type: LOAD_SECRETS_ERROR,
		error,
	}
}

export function postSecretToDatabase() {
	return {
		type: POST_SECRET,
	}
}