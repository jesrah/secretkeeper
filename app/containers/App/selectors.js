// import { createSelector } from 'reselect';

// const selectRoute = state => state.get('route');

// const makeSelectLocation = () =>
//   createSelector(selectRoute, routeState => routeState.get('location').toJS());

// export { makeSelectLocation };

import { createSelector } from 'reselect';

const selectGlobal = state => state.get('global');

const selectRoute = state => state.get('route');

const makeSelectCurrentSecret = () =>
  createSelector(selectGlobal, globalState =>
    globalState.getIn(['insert', 'currentSecret']),
  );

const makeSelectLoading = () =>
  createSelector(selectGlobal, globalState =>
    globalState.getIn(['secrets', 'loading']),
  );

const makeSelectInsertingToDb = () =>
  createSelector(selectGlobal, globalState =>
    globalState.getIn(['insert', 'isInserting']),
  );

const makeSelectError = () =>
  createSelector(selectGlobal, globalState =>
    globalState.getIn(['secrets', 'error']),
  );

const makeSelectDbError = () =>
  createSelector(selectGlobal, globalState =>
    globalState.getIn(['insert', 'error']),
  );

const makeSelectSecrets = () =>
  createSelector(selectGlobal, globalState =>
    globalState.getIn(['secrets', 'items']),
  );

const makeSelectLocation = () =>
  createSelector(selectRoute, routeState => routeState.get('location').toJS());

export {
  selectGlobal,
  makeSelectCurrentSecret,
  makeSelectLoading,
  makeSelectInsertingToDb,
  makeSelectError,
  makeSelectDbError,
  makeSelectSecrets,
  makeSelectLocation,
};
