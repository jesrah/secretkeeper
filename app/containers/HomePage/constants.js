export const SUBMIT_SECRET = 'secretkeeper/app/HomePage/SUBMIT_SECRET';