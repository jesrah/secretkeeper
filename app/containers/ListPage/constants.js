/*
 *
 * ListPage constants
 *
 */

export const GET_SECRETS = 'app/ListPage/GET_SECRETS';
