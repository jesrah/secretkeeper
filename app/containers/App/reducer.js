/*
 * AppReducer
 *
 * The reducer takes care of our data. Using actions, we can change our
 * application state.
 * To add a new action, add it to the switch statement in the reducer function
 *
 * Example:
 * case YOUR_ACTION_CONSTANT:
 *   return state.set('yourStateVariable', true);
 */

import { fromJS } from 'immutable';

import { LOAD_SECRETS_SUCCESS, LOAD_SECRETS, LOAD_SECRETS_ERROR } from './constants';

// The initial state of the App
const initialState = fromJS({
  secrets: {
    loading: false,
    error: false,
    items: [],
    load: true,
  },
});

//understand what's happening here
function appReducer(state = initialState, action) {
  switch (action.type) {
    case LOAD_SECRETS:
      return state
        .setIn(['secrets', 'loading'], true)
        .set(['secrets', 'error'], false)
        .set(['secrets', 'items'], false);
    case LOAD_SECRETS_SUCCESS:
      return state
        .setIn(['secrets', 'items'], action.secrets)
        .set(['secrets', 'loading'], false);
    case LOAD_SECRETS_ERROR:
      return state
        .setIn(['secrets', 'error'], action.error)
        .setIn(['secrets', 'loading'], false);
    default:
      return state;
  }
}

export default appReducer;
