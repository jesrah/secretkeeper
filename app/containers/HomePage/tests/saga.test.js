/**
 * Tests for HomePage sagas
 */

import { takeEvery } from 'redux-saga/effects';

import { POST_SECRET_DATABASE } from 'containers/App/constants';

import secretSubmissions, { insertSecret } from '../saga';

/* eslint-disable redux-saga/yield-effects */

describe('secretSubmissions Saga', () => {
  const secretSubmissionsSaga = secretSubmissions();

  it('should start task to watch for POST_SECRET_DATABASE action', () => {
    const takeEveryDescriptor = secretSubmissionsSaga.next().value;
    expect(takeEveryDescriptor).toEqual(takeEvery(POST_SECRET_DATABASE, insertSecret));
  });
});
