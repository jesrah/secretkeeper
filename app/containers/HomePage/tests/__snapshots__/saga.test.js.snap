// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`insertSecret Saga should call the secretLoadingError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`insertSecret Saga should call the secretLoadingError action if the response errors 2`] = `
Promise {
  "_a": undefined,
  "_c": Array [],
  "_d": false,
  "_h": 0,
  "_n": false,
  "_s": 0,
  "_v": undefined,
}
`;

exports[`insertSecret Saga should dispatch the secretsLoaded action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "SELECT": Object {
    "args": Array [],
    "selector": [Function],
  },
}
`;

exports[`insertSecret Saga should dispatch the secretsLoaded action if it requests the data successfully 2`] = `
Promise {
  "_a": undefined,
  "_c": Array [],
  "_d": false,
  "_h": 0,
  "_n": false,
  "_s": 0,
  "_v": undefined,
}
`;
